projects/packages/connection/src/class-manager.php

```diff
@@ -1922,9 +1923,44 @@ public function authorize( $data = array() ) {
 	/**
 	 * Disconnects from the Jetpack servers.
 	 * Forgets all connection details and tells the Jetpack servers to do the same.
+	 *
+	 * @param boolean $disconnect_wpcom Should disconnect_site_wpcom be called.
 	 */
-	public function disconnect_site() {
+	public function disconnect_site( $disconnect_wpcom = true ) {
+		wp_clear_scheduled_hook( 'jetpack_clean_nonces' );
+
+		( new Nonce_Handler() )->clean_all();
+
+		// If the site is in an IDC because sync is not allowed,
+		// let's make sure to not disconnect the production site.
+		if ( $disconnect_wpcom ) {
+			$tracking = new Tracking();
+			$tracking->record_user_event( 'disconnect_site', array() );
+
+			$this->disconnect_site_wpcom( true );
+		}
+
+		$this->delete_all_connection_tokens( true );
 
+		\Jetpack_Options::update_option( 'activated', 4 );
```

Comment:
I think that the `jetpack_activated` option might be specific to the Jetpack plugin, so we probably should only update it when Jetpack is active. Do you know if this option has any use if Jetpack is not active?

Reply:
I moved this back into Jetpack on the jetpack_idc_disconnect action.